lib/ansible/plugins/cache/__init__.py

```diff
@@ -298,18 +299,19 @@ def update(self, key, value):
         self._plugin.set(key, host_cache)
 
 
-class InventoryFileCacheModule(BaseFileCacheModule):
-    """
-    A caching module backed by file based storage.
-    """
-    def __init__(self, plugin_name, timeout, cache_dir):
-
-        self.plugin_name = plugin_name
-        self._timeout = timeout
+class InventoryCacheModule(FactCache, AnsiblePlugin):
```

Comment:
unsure of this, `FactCache` is really 'mapping cache'  and we call it directly, making it 'autoload' the actual cache plugin, we might just want to avoid this 'magic' and build the mapping methods into the basecache and call the plugins directly after they inherit.